src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -4051,6 +4078,410 @@ .maxstack  3
+  .maxstack  4
+  IL_0000:  ldc.i4.3
+  IL_0001:  conv.u
+  IL_0002:  sizeof     ""delegate*<int,int>""
+  IL_0008:  mul.ovf.un
+  IL_0009:  localloc
+  IL_000b:  dup
+  IL_000c:  ldftn      ""int C.Getter(int)""
+  IL_0012:  stind.i
+  IL_0013:  dup
+  IL_0014:  sizeof     ""delegate*<int,int>""
+  IL_001a:  add
+  IL_001b:  ldftn      ""int C.Getter(int)""
+  IL_0021:  stind.i
+  IL_0022:  dup
+  IL_0023:  ldc.i4.2
+  IL_0024:  conv.i
+  IL_0025:  sizeof     ""delegate*<int,int>""
+  IL_002b:  mul
+  IL_002c:  add
+  IL_002d:  ldftn      ""int C.Getter(int)""
+  IL_0033:  stind.i
+  IL_0034:  call       ""void C.Print(delegate*<int,int>*)""
+  IL_0039:  ret
+}
+");
+        }
+
+        [Fact]
+        public void FunctionPointerCannotBeUsedAsSpanArgument()
+        {
+            var comp = CreateCompilationWithSpan(@"
+using System;
+static unsafe class C
+{
+    static void Main()
+    {
+        Span<delegate*<int, int>> p = stackalloc delegate*<int, int>[1];
+    }
+}
+", options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.RegularPreview);
+
+            comp.VerifyDiagnostics(
+                // (7,14): error CS0306: The type 'delegate*<int,int>' may not be used as a type argument
+                //         Span<delegate*<int, int>> p = stackalloc delegate*<int, int>[1];
+                Diagnostic(ErrorCode.ERR_BadTypeArgument, "delegate*<int, int>").WithArguments("delegate*<int,int>").WithLocation(7, 14)
+            );
+        }
+
+        [Fact]
+        public void RecusivelyUsedTypeInFunctionPointer()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+namespace Interop
+{
+    public unsafe struct PROPVARIANT
+    {
+        public CAPROPVARIANT ca;
+    }
+    public unsafe struct CAPROPVARIANT
+    {
+        public uint cElems;
+        public delegate*<PROPVARIANT> pElems;
+        public delegate*<PROPVARIANT> pElemsProp { get; }
+    }
+}");
+        }
+
+        [Fact]
+        public void VolatileFunctionPointerField()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static volatile delegate*<void> ptr;
+    static void Print() => Console.Write(1);
+    static void Main()
+    {
+        ptr = &Print;
+        ptr();
+    }
+}", expectedOutput: "1");
+
+            verifier.VerifyIL("C.Main", expectedIL: @"
+{
+  // Code size       26 (0x1a)
+  .maxstack  1
+  IL_0000:  ldftn      ""void C.Print()""
+  IL_0006:  volatile.
+  IL_0008:  stsfld     ""delegate*<void> C.ptr""
+  IL_000d:  volatile.
+  IL_000f:  ldsfld     ""delegate*<void> C.ptr""
+  IL_0014:  calli      ""delegate*<void>""
+  IL_0019:  ret
+}
+");
+        }
+
+        [Fact]
+        public void SupportedBinaryOperators()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static (bool, bool, bool, bool, bool, bool) DoCompare(delegate*<void> func_1a, delegate*<string, void> func_1b)
```

Comment:
I didn't think it was necessary or useful to verify the exact IL for each type of binary operator conversion, so I split out the operators for just the function pointer to function pointer operators and verified this one function.